docs/enhancements/add-kernel-parameters.md

```diff
@@ -0,0 +1,128 @@
+### User Stories
+
+#### Story 1
+
+When having slow NIC, increase the timeout to have the IP address configured.
+
+### Implementation Details/Notes/Constraints
+
+#### iPXE implementation
+
+iPXE implementation should be quite straight forward.  The arguments will be concatenated at the end of the 
+**kernel** command which is part of the iPXE script.
+
+#### Minimal and full ISO
+
+When streaming the discovery ISO file, the kernel arguments are included
+as part of some files: grub.cfg and isolinux.cfg
+
+Each of these files contains an area where the kernel arguments can be
+embedded. This can be identified by the regex **"(\n#+) COREOS_KARG_EMBED_AREA"**.
+The arguments should overwrite this are from the beginning and should
+not pass the last # (hash sign). The added kernel will be injected in the beginning.
+These arguments will be seperated by space **(" ")**.  When embedding the kernel arguments,
+the size of the above files will not change.
+
+Starting from version 4.10, there is a file that contains the file names,
+while in older versions the file does not exist. If the file exists, the
+files to embed the kernel arguments are taken from this config file.
+Otherwise, the default files **[/EFI/redhat/grub.cfg, /isolinux/isolinux.cfg]**
+are used.
+
+
+We will need the image service to query the assisted service to get the kernel arguments.
+This will require authentication.  The authentication methodology may have 2 variants:  
+
+1. When running in cloud
+2. When running in disconnected environment
+
+The image service already queries the assisted service.  We will use the same authentication 
+methods.  We will need to verify that the infra-env endpoint may be accessed using the same authentication
+mechanism.
+
+#### UI
+
+Eventually, UI may support kernel arguments modification in all places that infra-env is manipulated.
+#### Kube API
+
+The discovery kernel arguments should be defined in the spec section in InfraEnv CR.  
+They will have the following format:
+
+In InfraEnv Spec section:
+
+```sh
+DiscoveryKernelArguments []KernelArgument `json:"discoveryKernelArguments,omitempty"`
+```
+
+In addition there is the definition of **KernelArgument**:
+
+```sh
+type KernelArgument struct {
+
+	Operation string `json:"operation,omitempty"`
+
+	Value string `json:"value,omitempty"`
+}
+```
+
+
+
+### Risks and Mitigations
+
+## Design Details [optional]
+
+### Open Questions
+Coreos Installer already supports the operations mentioned in this document but does not 
+have full-support for streaming.  Once it will have such support, the image service will move to 
+use the Coreos Installer.  It will be done not as part of the effort to add discovery kernel parameters.
+
+Here is a link to the Coreos Installer kernel arguments modification reference:
+https://coreos.github.io/coreos-installer/cmd/iso/#coreos-installer-iso-kargs-modify
+### UI Impact
+
```

Comment:
SaaS UI:
- What validations should be performed before sending the kernel args?
- If users modify the kernel args, will they be required to discover the hosts again?